memorious/operations/parse.py

```diff
@@ -91,15 +92,52 @@ def parse_for_metadata(context, data, html):
     return data
 
 
+def get_entity_id_from_keys(keys, html):
+    key_string = ""
+    entity_id = ""
+    if isinstance(keys, list):
+        for item in keys:
+            temp_key = html.xpath(item["key"])
+            if len(temp_key) > 0:
+                key_string += "".join(temp_key)
+
+        if (
+            hashlib.md5(key_string.encode("utf-8")).hexdigest()
+            != hashlib.md5("".encode("utf-8")).hexdigest()
+        ):
+            entity_id = hashlib.md5(key_string.encode("utf-8")).hexdigest()
+            log.info("Entity id generated from keys: {}".format(entity_id))
+            return entity_id
+        else:
+            log.warn(
+                "Keys generated an empty string. There may be something wrong with the xpath"
+            )
+
+        return
+
+
 def parse_ftm(context, data, html):
     properties = context.params.get("properties")
     properties_dict = {}
+
     for key, value in properties.items():
         properties_dict[key] = html.xpath(value)
 
+    data["entity_id"] = hashlib.md5(data["url"].encode("utf-8")).hexdigest()
```

Comment:
It's probably better and more consistent to make keys out of the supplied keys only instead of falling back on a default. We should instead raise an error in case of null keys so that the user can notice and change their keying strategy.